tests/ttnn/unit_tests/operations/test_moreh_dot.py

```diff
@@ -16,84 +16,52 @@
 )
 
 
-def create_tt_tensor(tensor, layout, dtype):
-    return ttnn.from_torch(tensor, layout=layout, dtype=dtype)
-
-
 def get_torch_dtype(dtype):
     if dtype == ttnn.int32:
         return torch.int32
-    elif dtype == ttnn.float32:
-        return torch.float32
     else:
         return torch.bfloat16
 
 
-def get_tensors(
-    input_shape,
-    other_shape,
-    output_shape,
-    require_input_grad,
-    require_other_grad,
-    is_1d,
-    device,
-    npu_dtype,
-    cpu_dtype,
-    use_randint=True,
-):
-    npu_layout = ttnn.TILE_LAYOUT
-    cpu_layout = ttnn.ROW_MAJOR_LAYOUT
+def run_moreh_dot_test(input_shape, npu_dtype, device, optional_output=False):
+    # TODO @thanhnguyen-moreh: Support bfloat8_b in kernel
+    if npu_dtype == ttnn.bfloat8_b:
+        pytest.skip(f"bfloat8_b is not supported in the kernel")
+    cpu_dtype = get_torch_dtype(npu_dtype)
+    output_shape = [1, 1, 1, 1]
 
-    # create tensors for forward
-    if use_randint:
-        input = torch.randint(-2, 3, input_shape, dtype=torch.bfloat16)
-        other = torch.randint(-2, 3, other_shape, dtype=torch.bfloat16)
-        output = torch.randint(-2, 3, output_shape, dtype=torch.bfloat16)
+    if npu_dtype == ttnn.int32:
+        torch_input = torch.randint(-2, 3, input_shape, dtype=cpu_dtype)
+        torch_other = torch.randint(-2, 3, input_shape, dtype=cpu_dtype)
     else:
-        input = torch.rand(input_shape, dtype=torch.bfloat16)
-        other = torch.rand(other_shape, dtype=torch.bfloat16)
-        output = torch.rand(output_shape, dtype=torch.bfloat16)
+        torch_input = torch.rand(input_shape, dtype=cpu_dtype)
+        torch_other = torch.rand(input_shape, dtype=cpu_dtype)
 
-    # inputs must be of type bfloat16 or bfloat8
-    tt_input = create_tt_tensor(input, cpu_layout, ttnn.bfloat16).pad_to_tile(float(1)).to(npu_layout).to(device)
-    tt_other = create_tt_tensor(other, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
-    tt_output = create_tt_tensor(output, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
+    tt_input = ttnn.from_torch(torch_input, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
+    tt_other = ttnn.from_torch(torch_other, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
```

Comment:
Wrong logic. Should use ttnn_dtype here.
Throughout, use torch_dtype and ttnn_dtype.